### ISSUE TITLE
Minecraft 1.2.5 无法启动

### ISSUE THREAD

**Issue by MinimalMio** [FCL-Team/FoldCraftLauncher#730]
### Before making issue / 在发布提问之前...

- [X] I have read the content above.我已经阅读了上方的内容。
- [X] I understand what I'm asking is not about modpack.我明白我询问的不是整合包相关的问题。
- [X] I have searched for existing issues.我已经搜索了现有的 issues。
- [X] I have tried switching renderer.我已经尝试过切换渲染器。

### Launcher Version / 启动器版本

1.1.9

### Device Information / 设备信息

Redmi Note 7, MIUI 12.5.7, Android 10, arm64

### Problem Description / 问题描述

启动Minecraft 1.2.5时，不能正常启动（如下）
（截图1：Boat模式下，卡死在“Mojang”徽标界面）
![Screenshot_2024-11-25-19-37-32-761_com tungsten fcl](https://github.com/user-attachments/assets/2cba342d-73b4-4362-babf-0675147e55cc)
（截图2：Pojav模式下，无法正常启动）
![Screenshot_2024-11-25-19-37-11-477_com tungsten fcl](https://github.com/user-attachments/assets/d5cf0d0d-7778-4071-8497-9347eccf0b73)
此外，该bug还影响到了Minecraft 1.1，以及1.3~1.5.2的所有版本

### Java

Java 8

### Renderer / 渲染器

Holy-GL4ES

### Game Version / 游戏版本

1.2.5

### Game Crash Report / 游戏崩溃报告

它并没有崩溃，而是卡死

### Launcher Crash Report / 启动器崩溃报告

它并没有崩溃

### Launcher Log File / 启动器日志文件

[logs.zip](https://github.com/user-attachments/files/17902369/logs.zip)

**Comment by ShirosakiMio** [comment]
![Screenshot_20241126_095645_com tungsten fcl](https://github.com/user-attachments/assets/67bc12f7-ef35-4371-bc6f-9fa0fe7f5de9)
我这不开版本隔离的话跟你图例一样，开了之后是正常进去了

**Comment by MinimalMio** [comment]
我这边开不开版本隔离都一样

________________________________

**Comment by u3560174892** [comment]
I found why this happens and can reproduce it.

Short answer : Launcher is downloading wrong options.txt file for 1.1 ~ 1.5.2 (Seems like it is for higher versions? file size is x3 bloated)

Long answer : 
1. Delete options.txt file on minecraft folder
2. try launch 1.1~1.5.2 version (Do not run other version)
3. Reproduced


So solve way is
1. launch 1.0 version and 
2. just copy options.txt
3. then run 1.1~1.5.2 version once, if crashed then close
4. paste copied options.txt
5. now game can run well


So maybe you should fix download links for options.txt file for 1.1~1.5.2

**Comment by ShirosakiMio** [comment]
> I found why this happens and can reproduce it.
> 
> Short answer : Launcher is downloading wrong options.txt file for 1.1 ~ 1.5.2 (Seems like it is for higher versions? file size is x3 bloated)
> 
> Long answer :
> 
> 1. Delete options.txt file on minecraft folder
> 2. try launch 1.1~1.5.2 version (Do not run other version)
> 3. Reproduced
> 
> So solve way is
> 
> 1. launch 1.0 version and
> 2. just copy options.txt
> 3. then run 1.1~1.5.2 version once, if crashed then close
> 4. paste copied options.txt
> 5. now game can run well
> 
> So maybe you should fix download links for options.txt file for 1.1~1.5.2

Yes, I have verified it as well. The inability to launch the April Fools' Day version 2.0 is also caused by this issue. Thank you for your attempt. The options.txt file is built-in, and if no options.txt file is detected in the directory, the launcher will create one (it appears to be using the version 1.7.10+). I will try to prevent the launcher from creating this file when running versions below 1.5.2.

**Comment by u3560174892** [comment]
Tested with actions build, and confirmed this fixed for clean installing 1.1 ~ 1.5.2 version.

But there is still issue. Turn off Isolate Working directory and then run 1.12.2 to make options.txt file, then try to run 1.1 ~ 1.5.2 version, game still crash.

When I delete options.txt file from minecraft folder manually and then run 1.1 ~ 1.5.2, then it can be fix.

Seems like some launcher fixed this issue, but is it hard to fix on FCL?

**Comment by u3560174892** [comment]
I think I found why it crashes happen and can reproduce.

For 1.1~1.5.2 options.txt file, if there is no Capital letters on "lang:" line(ex. en_"US"), then game crashes.


So possibly can be fix by:
1. when try to launch 1.1~1.5.2,
2. launcher checks options.txt file and
3. if there is no Capital letters on "lang:" line,
4. then remove options.txt file

**Comment by ShirosakiMio** [comment]
> Tested with actions build, and confirmed this fixed for clean installing 1.1 ~ 1.5.2 version.
> 
> But there is still issue. Turn off Isolate Working directory and then run 1.12.2 to make options.txt file, then try to run 1.1 ~ 1.5.2 version, game still crash.
> 
> When I delete options.txt file from minecraft folder manually and then run 1.1 ~ 1.5.2, then it can be fix.
> 
> Seems like some launcher fixed this issue, but is it hard to fix on FCL?

[options.txt](https://github.com/AngelAuraMC/Amethyst-Android/blob/5a797bfe611f73f7bf8b9d3bb544e61398ea3c08/app_pojavlauncher/src/main/assets/options.txt#L20)

[options.txt](https://github.com/FCL-Team/FoldCraftLauncher/blob/main/FCL/src/main/assets/options.txt#L20)

**Comment by ShirosakiMio** [comment]
https://github.com/FCL-Team/FoldCraftLauncher/actions/runs/20703107436
I have added a more reasonable fix method. Each time it starts, it checks the game version. If it is less than 1.11, the text after "lang:" will be modified by converting "xx_xx" to "xx_XX", which works for any language. Of course, if the version is greater than 1.11, the reverse modification will also be applied.

**Comment by u3560174892** [comment]
It works now!

But there is one more problem, if update 1.1 ~ 1.5.2 version to 1.17 or higher with default minecraft settings, minecraft automatically adds mipmapLevels: 4 on config.yml and if I try to run with Krypton Wrapper, it crashes on start. 

that mipmapLevels setting 1 or higher has problem with all default renderers that currently have on foldcraftlauncher.(KW and Zink instantly crash, VirGL has some issue with rendering) I added LTW and it works without any problem.

It seems like it is renderer fault. So I think most easy way to solve is changing default renderer to LTW for 1.17+, because LTW library only consume 1MB.
